hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/multitenant/TestRangerBGSyncService.java

```diff
@@ -0,0 +1,688 @@
+    } catch (Exception e) {
+      Assert.fail(e.getMessage());
+    }
+
+    try {
+      AccessPolicy tenant1BucketCreatePolicy = newBucketAccessPolicy(
+          volumeName, tenantId);
+      LOG.info("Creating BucketAccess policy in Ranger: {}",
+          tenant1BucketCreatePolicy.getPolicyName());
+      auth.createAccessPolicy(tenant1BucketCreatePolicy);
+      policiesCreated.add(tenant1BucketCreatePolicy.getPolicyName());
+    } catch (Exception e) {
+      Assert.fail(e.getMessage());
+    }
+  }
+
+  public void cleanupPolicies() {
+    for (String name : policiesCreated) {
+      try {
+        LOG.info("Deleting policy: {}", name);
+        auth.deletePolicyByName(name);
+      } catch (Exception e) {
+        LOG.error(e.getMessage());
+      }
+    }
+  }
+
+  public void cleanupRoles() {
+    for (String roleObj : rolesCreated) {
+      final JsonObject jObj = new JsonParser().parse(roleObj).getAsJsonObject();
+      final String roleId = jObj.get("id").getAsString();
+      final String roleName = jObj.get("name").getAsString();
+      try {
+        LOG.info("Deleting role: {}", roleName);
+        auth.deleteRole(roleId);
+      } catch (Exception e) {
+        LOG.error(e.getMessage());
+      }
+    }
+  }
+
+  public void cleanupUsers() {
+    for (BasicUserPrincipal user : usersCreated) {
+      try {
+        LOG.info("Deleting user: {}", user);
+        String userId = auth.getUserId(user);
+        auth.deleteUser(userId);
+      } catch (Exception e) {
+        LOG.error(e.getMessage());
+      }
+    }
+  }
+
+  public void cleanupOMDB() {
+    try {
+      omMetadataManager.getTenantStateTable().delete(TENANT_ID);
+      omMetadataManager.getTenantAccessIdTable().delete(
+          OMMultiTenantManager.getDefaultAccessId(TENANT_ID, USER_ALICE_SHORT));
+      omMetadataManager.getTenantAccessIdTable().delete(
+          OMMultiTenantManager.getDefaultAccessId(TENANT_ID, USER_BOB_SHORT));
+    } catch (IOException e) {
+      LOG.error(e.getMessage());
+    }
+  }
+
+  public void cleanupPoliciesRolesUsers() {
+    cleanupPolicies();
+    cleanupRoles();
+    cleanupUsers();
+
+    cleanupOMDB();
+  }
+
+  /**
+   * OM DB does not have the tenant state.
+   * Expect sync service to clean up all the leftover multi-tenancy
+   * policies and roles in Ranger.
+   */
+  @Test
+  public void testRemovePolicyAndRole() throws Exception {
+    long startingRangerVersion = initBGSync();
+
+    // Create roles and policies in ranger that are NOT
+    // backed up by OzoneManger Multi-Tenant tables
+    createRolesAndPoliciesInRanger(false);
+    long rangerVersionAfterCreation = bgSync.getLatestRangerServiceVersion();
+    Assert.assertTrue(rangerVersionAfterCreation >= startingRangerVersion);
+
+    bgSync.start();
+    // Wait for sync to finish once.
+    // The counter is incremented at the beginning of the run, hence the ">"
+    GenericTestUtils.waitFor(
+        () -> bgSync.getRangerSyncRunCount() > 1L, 1000, 60000);
```

Comment:
We can use some multiple of `TEST_SYNC_INTERVAL_SEC` and `TEST_SYNC_TIMEOUT_SEC` here instead of magic numbers.